src/tuntap/tun.go

```diff
@@ -170,9 +166,11 @@ func (tun *TunAdapter) _start() error {
 	}
 	tun.core.SetMaximumSessionMTU(tun.MTU())
 	tun.isOpen = true
-	go tun.handler()
 	tun.reader.Act(nil, tun.reader._read) // Start the reader
 	tun.ckr.init(tun)
+	tun.packetConn.SetReadCallback(func(addr net.Addr, bs []byte) {
```

Comment:
We should probably commit using the `PacketConn` interface at first, at least to test things, before we insert callbacks etc to work around it / make it actor-friendly. Or commit to using actors and don't worry about matching the `PacketConn` interface at all for now. Either way, mixing and matching is likely to hide bugs in one interface or the other.

Reply:
Indeed - this was mostly just to satisfy the existing interface rather than breaking it, but it wouldn't be too difficult to change.